contrib/librdkafka-cmake/CMakeLists.txt

```diff
@@ -8,22 +8,23 @@ endif()
 set(RDKAFKA_SOURCE_DIR "${ClickHouse_SOURCE_DIR}/contrib/librdkafka/src")
 
 set(SRCS
+  "${RDKAFKA_SOURCE_DIR}/cJSON.c"
   "${RDKAFKA_SOURCE_DIR}/crc32c.c"
-#  "${RDKAFKA_SOURCE_DIR}/lz4.c"
-#  "${RDKAFKA_SOURCE_DIR}/lz4frame.c"
-#  "${RDKAFKA_SOURCE_DIR}/lz4hc.c"
+#  "${RDKAFKA_SOURCE_DIR}/lz4.c"      # WITH_LZ4_EXT
+#  "${RDKAFKA_SOURCE_DIR}/lz4frame.c" # WITH_LZ4_EXT
+#  "${RDKAFKA_SOURCE_DIR}/lz4hc.c"    # WITH_LZ4_EXT
   "${RDKAFKA_SOURCE_DIR}/rdaddr.c"
   "${RDKAFKA_SOURCE_DIR}/rdavl.c"
   "${RDKAFKA_SOURCE_DIR}/rdbuf.c"
   "${RDKAFKA_SOURCE_DIR}/rdcrc32.c"
-  "${RDKAFKA_SOURCE_DIR}/rddl.c"
+  #"${RDKAFKA_SOURCE_DIR}/rddl.c" #  WITH_LIBDL OR WIN32 - we don't want to support dynamic loading
```

Comment:
@filimonov This fixes the issue with `alloca()` on freebsd, and made the build work even without https://github.com/ClickHouse/librdkafka/pull/7

Reply:
But I would still backport the fix (it is a shame that I did not did it by myself)

Reply:
+1. Great, so i think we can go with the upstream branch without extra patches.